### DISCUSSION TITLE
Does rhf field names have a escape character?

### DISCUSSION THREAD

**Discussion by GustavoMelloGit** [react-hook-form/react-hook-form#10097] [Q&A] [answered]
## The Problem
My project uses [Sequelize Search Builder](https://github.com/segemun/sequelize-search-builder) to filter backend data. The issue is that my field should ideally have the same name as the filter, but it's not possible due to how the library builds the query. (Ex: filter[name]="Joseph")

## Solution
Would be great if i could name my fields as "filter[name]" without actually triggering a array

**Comment by leapful** [comment, marked as answer]
@GustavoMelloGit 

It seems that you are mis-understanding the documentation of library `sequelize-search-builder`. The pattern in example `filter[name]=John&filter[surname]=Smith` is the query params from API url with HTTP method GET, like: `https://yoursite.com/search?filter[name]=John&filter[surname]=Smith`.

So the problem is not from `react-hook-form` or `sequelize-search-builder`, it's from your code implementation to build the correct query params by using form values.

For example:

```
const { handleSubmit } = useForm();

const onSubmit = (formValues) => {
    // Assume formValues = { name: 'John', surname: 'Smith' }, just build the query params for API '/search'
   const queryParams = Object.entries(formValues).map(([key, value]) => {
      return `filter[${key}]=${value}`;
    });

   fetch(`/search?${queryParams.join('&')}`);
}

return <form onSubmit={handleSubmit(onSubmit)} >
...
</form>
```
